assets/js/blocks/product-category/block.js

```diff
@@ -85,23 +120,24 @@ class ProductByCategoryBlock extends Component {
 
 	getInspectorControls() {
 		const { attributes, setAttributes } = this.props;
-		const { columns, catOperator, contentVisibility, editMode, orderby, rows } = attributes;
+		const { isEditing } = this.state;
+		const { columns, catOperator, contentVisibility, orderby, rows } = attributes;
 
 		return (
 			<InspectorControls key="inspector">
 				<PanelBody
 					title={ __( 'Product Category', 'woo-gutenberg-products-block' ) }
-					initialOpen={ ! attributes.categories.length && ! editMode }
+					initialOpen={ ! attributes.categories.length && ! isEditing }
 				>
 					<ProductCategoryControl
 						selected={ attributes.categories }
 						onChange={ ( value = [] ) => {
 							const ids = value.map( ( { id } ) => id );
-							setAttributes( { categories: ids } );
+							this.setAttributes( { categories: ids } );
 						} }
 						operator={ catOperator }
 						onOperatorChange={ ( value = 'any' ) =>
-							setAttributes( { catOperator: value } )
+							this.setAttributes( { catOperator: value } )
```

Comment:
This is calling my local state based setAttributes method. Am I calling this correctly? It works anyhow.

Reply:
It looks fine. As I said in another comment, I would change its name to avoid confusion with `setAttributes` prop and to make clear we are only updating the temporary attributes, but the calling looks good to me. :+1: